### PR TITLE
[Feature] Moved to Gorilla/mux for server routing

### LINKED ISSUE #1410: Move to Gorilla/mux for server routing
**Current Behavior**

We are using **HTTProuter** for routing & serving our UI pages.

**Desired Behavior**

Use **Gorilla/mux** for server routing, remove complex network logic from handlers and use mux UI handlers to server SSR next UI bundle.

---
**Resources**

Read [this](https://www.nicolasmerouze.com/guide-routers-golang). article for a brief Comparision between HTTProuter & gorilla/mux.

### PR DESCRIPTION
**Pull request by 0xfirefist**
**Description**

This PR fixes #1410

**Notes for Reviewers**
- Changed the http.ServeMux to mux.Router
- The code which was verifying request method is now added to the mux itself and is commented in handlers.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: router/server.go [hunk 1]
```diff
@@ -102,23 +133,25 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
...
 	}))
 
-	mux.Handle("/", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		handlers.ServeUI(w, r, "", "../ui/out/")
-	}))))
+	// gMux.Handle("/", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	// 	handlers.ServeUI(w, r, "", "../ui/out/")
+	// }))))
+
+	gMux.Methods("GET").PathPrefix("/").Handler(h.ProviderMiddleware(h.AuthMiddleware(http.StripPrefix("/", http.FileServer(http.Dir("../ui/out/"))))))
```
Comment (kushthedude) [inline router/server.go:151]: We would still want to use the UI handler to serve our SPA, use the handler SPA method

**Review by kushthedude — COMMENTED** [review]
Great work @kalradev ,most of the stuff looks good, we would want to decouple the handlers more and remove the branching as much as can be done. 
Please restore the UI Handler rather than providing the stripped output path in the router itself, I guess gorilla/mux would utilise most of the code written in our UI handler.

**Comment by kalradev** [comment]
Okay I will made all these changes and will think of how to decouple and how to reduce branching.

**Comment by leecalcote** [comment]
@kalradev a conflict to resolve - https://github.com/layer5io/meshery/pull/1439/conflicts

**Comment by kushthedude** [comment]
> @kalradev a conflict to resolve - https://github.com/layer5io/meshery/pull/1439/conflicts

@kalradev In my recent PR, I renamed `promotheus/scan` to `prometheus/scan`, please resolve the conflict with updating the scan endpoint.

**Pull request merged**